lighthouse-core/audits/byte-efficiency/uses-optimized-images.js

```diff
@@ -80,7 +80,7 @@ class UsesOptimizedImages extends ByteEfficiencyAudit {
     const warnings = [];
     for (const image of images) {
       if (image.failed) {
-        warnings.push(`Unable to decode ${URL.getURLDisplayName(image.url)}`);
+        warnings.push(`Unable to decode ${URL.getURLDisplayName(image.url)}: ${image.errMsg}`);
```

Comment:
I believe these messages used to be included but were removed because they typically had nothing to do with the user.

is the goal for *us* to know why they failed? because we should already be tracking those in Sentry via the gatherer :)

Reply:
> is the goal for us to know why they failed? because we should already be tracking those in Sentry via the gatherer :)

yeah. but what about when i run LH? I am forced to print my woes in devtools, since I can't access the artifacts (and we don't log these warnings anywhere).

but it's also nice to be able to see from a user report (which most often comes in the form of a PNG) what the error was

Reply:
> but what about when i run LH? 

you're not really the target user/reason we would hide the error :) I mean other than you, who is going to be able to do anything with this additional string?

> but it's also nice to be able to see from a user report (which most often comes in the form of a PNG) what the error was

true! but when has this last happened where we've gotten a screenshot with warnings filed we needed to investigate? :) we [already know what errors can happen here](https://sentry.io/organizations/google-lighthouse/issues/?project=174697&query=is%3Aunresolved+gatherer%3AOptimizedImages&statsPeriod=14d) and we don't really do much about them (existing set seems just intentionally ignored target, response was evicted from memory, or wasn't a real image). 

I think my overall rationale here is the same reason we designed the error drawer, outside of the base warning that something was up, detailed error messages are generally useless information to the user and exist purely for facilitating bug reports. In this case, we don't seem to learn anything new and until the drawer is implemented I'm not super excited to *increase* the verbosity of that messaging.

We've already got other walls of text warnings like you've filed, so it's not like we're doing great in this arena already, but I'd like to avoid heading in the other direction if we can.

That's my spiel, but happy to move this forward too if you're unconvinced and would like to hunt these down anyway :)

Reply:
this all sounds good. I'll just add a log.error in the gathering when it happens.